lsat/interceptor.go

```diff
@@ -105,38 +114,72 @@ func (i *Interceptor) UnaryInterceptor(ctx context.Context, method string,
 	i.lock.Lock()
 	defer i.lock.Unlock()
 
-	addLsatCredentials := func(token *Token) error {
-		macaroon, err := token.PaidMacaroon()
+	// Create the context that we'll use to initiate the real request. This
+	// contains the means to extract response headers and possibly also an
+	// auth token, if we already have paid for one.
+	iCtx, err := i.newInterceptContext(ctx, opts)
+	if err != nil {
+		return err
+	}
+
+	// Try executing the call now. If anything goes wrong, we only handle
+	// the LSAT error message that comes in the form of a gRPC status error.
+	rpcCtx, cancel := context.WithTimeout(ctx, i.callTimeout)
+	defer cancel()
+	err = invoker(rpcCtx, method, req, reply, cc, iCtx.opts...)
+	if isPaymentRequired(err) {
```

Comment:
Nit: check `!isPaymentRequired` instead to save on indentation for the fragment below.

Reply:
Yes, good idea.